continuous-testing-foundations/dotnet/Pragmatest.Wallets/test/Pragmatest.Wallets.Web.IntegrationTests/WalletControllerIntegrationTests.cs

```diff
@@ -191,65 +214,78 @@ public async Task Withdraw_Withdraw5_Returns10Async()
+            //// Act  
 
-            // Act  
             HttpResponseMessage response = await client.PostAsync(endpoint, content);
 
-            // Assert
+            //// Assert
+            
             Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+            
             walletServiceMock.VerifyNoOtherCalls();
         }
 
         [Fact]
         public async Task Withdrawal_WithdrawalInsufficientBalanceException_ReturnsBadRequest()
         {
-            // Arrange
+            //// Arrange
+            
             int withdrawalAmount = 1000;
-            string endpoint = "Wallet/Withdraw";
-
+            
+            // Setup Mocks
+                        
             Mock<IWalletService> walletServiceMock = new Mock<IWalletService>();
 
             Withdrawal withdrawal = new Withdrawal { Amount = withdrawalAmount };
-            CompareLogic compare = new CompareLogic();
-
             walletServiceMock
-             .Setup(walletService => walletService.WithdrawFundsAsync(It.Is<Withdrawal>(actualWithdrawal => compare.Compare(withdrawal, actualWithdrawal).AreEqual)))
+             .Setup(walletService => walletService.WithdrawFundsAsync(It.Is<Withdrawal>(
+                 actualWithdrawal => _comparer.Compare(withdrawal, actualWithdrawal).AreEqual)))
              .Throws<InsufficientBalanceException>();
 
             IWalletService walletService = walletServiceMock.Object;
-            
+
+            // Initialize HTTP client and request data
+
             WebApplicationFactory<Startup> factory = new CustomWebApplicationFactory<Startup>(services =>
                 services.SwapTransient(provider => walletService)
             );
 
-            WithdrawalRequest withdrawalRequest = new WithdrawalRequest { Amount = withdrawalAmount };
+            HttpClient client = factory.CreateClient();
 
-            StringContent content = new StringContent(JsonConvert.SerializeObject(withdrawalRequest), Encoding.UTF8, "application/json");
+            string endpoint = "Wallet/Withdraw";
+            WithdrawalRequest withdrawalRequest = new WithdrawalRequest { Amount = withdrawalAmount };
+            StringContent content = withdrawalRequest.AsStringContent();
 
-            HttpClient client = factory.CreateClient();
+            //// Act  
 
-            // Act  
             HttpResponseMessage response = await client.PostAsync(endpoint, content);
 
-            // Assert
+            //// Assert
+            
             Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
-            walletServiceMock.Verify(walletService => walletService.WithdrawFundsAsync(It.Is<Withdrawal>(actualWithdrawal => compare.Compare(withdrawal, actualWithdrawal).AreEqual)), Times.Once);
+            
+            walletServiceMock.Verify(walletService => walletService.WithdrawFundsAsync(It.Is<Withdrawal>(
+                    actualWithdrawal => _comparer.Compare(withdrawal, actualWithdrawal).AreEqual)
```

Comment:
Shall you make this expression common since you are using it a number of times?

Reply:
we could but it would not change much as it's already a simple function call.
it would change from: 
actualWithdrawal => _comparer.Compare(withdrawal, actualWithdrawal).AreEqual
to:
actualWithdrawal => compareWithdrawals(withdrawal,actualWithdrawal)